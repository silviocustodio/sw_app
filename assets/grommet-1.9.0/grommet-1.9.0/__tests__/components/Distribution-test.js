// (C) Copyright 2014-2016 Hewlett Packard Enterprise Development LP

import React from 'react';
import renderer from 'react-test-renderer';

import Distribution from '../../src/js/components/Distribution';

// needed because this:
// https://github.com/facebook/jest/issues/1353
jest.mock('react-dom');

describe('Distribution', () => {
  it('has correct default options', () => {
    const component = renderer.create(
      <Distribution series={[
        {"label": "First", "value": 40, "colorIndex": "graph-1"},
        {"label": "Second", "value": 30, "colorIndex": "accent-2"},
        {"label": "Third", "value": 20, "colorIndex": "unset"},
        {"label": "Fourth", "value": 10, "colorIndex": "graph-1"}
      ]} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
