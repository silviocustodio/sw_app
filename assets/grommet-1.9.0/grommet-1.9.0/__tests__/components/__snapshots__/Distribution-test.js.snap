// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Distribution has correct default options 1`] = `
<div
  className="grommetux-distribution grommetux-distribution--medium grommetux-distribution--loading"
>
  <svg
    aria-label="Loading"
    className="grommetux-distribution__graphic"
    onBlur={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    preserveAspectRatio="none"
    role="img"
    tabIndex="0"
    viewBox="0 0 400 200"
  >
    <rect
      className="grommetux-distribution__background"
      height={200}
      stroke="none"
      width={400}
      x={0}
      y={0}
    />
    <g>
      <path
        className="grommetux-distribution__loading-indicator grommetux-color-index-loading"
        d="M0,100 L400,100"
        stroke="none"
      />
    </g>
  </svg>
  <div
    aria-hidden={true}
    className="grommetux-distribution__labels"
    role="presentation"
  />
</div>
`;
